test/node/utf8_tests.js

```diff
@@ -0,0 +1,343 @@
+'use strict';
+
+const { Buffer } = require('buffer');
+const BSON = require('../register-bson');
+const { isNode6, isBrowser } = require('./tools/utils');
+const BSONError = BSON.BSONError;
+
+describe('UTF8 validation', function () {
+  // Test both browser shims and node which have different replacement mechanisms
+  const replacementChar = isNode6() || isBrowser() ? '\u{FFFD}\u{FFFD}\u{FFFD}' : '\u{FFFD}';
+  const replacementString = `hi${replacementChar}bye`;
+  const twoCharReplacementStr = `${replacementChar}${replacementChar}bye`;
+  const sampleValidUTF8 = BSON.serialize({
+    a: '😎',
+    b: 'valid utf8',
+    c: 12345
+  });
+
+  it('should throw error if true and false mixed for validation option passed in with valid utf8 example', function () {
+    const mixedTrueFalse1 = { validation: { utf8: { a: false, b: true } } };
+    const mixedTrueFalse2 = { validation: { utf8: { a: true, b: true, c: false } } };
+    expect(() => BSON.deserialize(sampleValidUTF8, mixedTrueFalse1)).to.throw(
+      BSONError,
+      'Invalid UTF-8 validation option - keys must be all true or all false'
+    );
+    expect(() => BSON.deserialize(sampleValidUTF8, mixedTrueFalse2)).to.throw(
+      BSONError,
+      'Invalid UTF-8 validation option - keys must be all true or all false'
+    );
+  });
+
+  it('should correctly handle validation if validation option contains all T or all F with valid utf8 example', function () {
+    let allTrue = { validation: { utf8: { a: true, b: true, c: true } } };
+    let allFalse = { validation: { utf8: { a: false, b: false, c: false, d: false } } };
+    expect(() => BSON.deserialize(sampleValidUTF8, allTrue)).to.not.throw();
+    expect(() => BSON.deserialize(sampleValidUTF8, allFalse)).to.not.throw();
+  });
+
+  it('should throw error if empty utf8 validation option passed in', function () {
```

Comment:
I think we're missing a test case for when a non-boolean validation option is specified for a key